test/tstQueryTreeCallbacksRay.cpp

```diff
@@ -0,0 +1,156 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+#include <numeric>
+#include <vector>
+
+#include "Search_UnitTestHelpers.hpp"
+// clang-format off
+#include "ArborXTest_TreeTypeTraits.hpp"
+// clang-format on
+
+template <typename DeviceType>
+struct NearestBoxToRay
+{
+  Kokkos::View<ArborX::Experimental::Ray *, DeviceType> rays;
+  int k;
+};
+
+template <typename DeviceType>
+struct BoxesIntersectedByRay
+{
+  Kokkos::View<ArborX::Experimental::Ray *, DeviceType> rays;
+};
+
+template <typename DeviceType>
+struct Boxes
+{
+  Kokkos::View<ArborX::Box *, DeviceType> elements;
+};
+
+namespace ArborX
+{
+template <typename DeviceType>
+struct AccessTraits<Boxes<DeviceType>, ArborX::PrimitivesTag>
+{
+  using memory_space = typename DeviceType::memory_space;
+  static KOKKOS_FUNCTION int size(Boxes<DeviceType> const &boxes)
+  {
+    return boxes.elements.size();
+  }
+  static KOKKOS_FUNCTION auto get(Boxes<DeviceType> const &boxes, int i)
+  {
+    return boxes.elements(i);
+  }
+};
```

Comment:
Why did you define `Boxes` with your own access traits?  We support `Kokkos::View<T*>` natively.